### PR TITLE
Replace native clients

### PR DESCRIPTION
**Pull request by agocke**
Same pull request as before, this one just bases against my previous branch as opposed to master to avoid duplicate commits and comments. Following is the text from the original PR:

This change continues on the trend of removing the native client code for the compiler server and replacing it with managed code. This is the PR which actually removes the native code! :) This commit deletes all of the native projects and replaces them with the managed client implementation, keeping exactly the same functionality as the old csc2/vbc2 projects.

As before, this PR builds on the commits of the previous one, namely #866, so that one should be reviewed and merged first.

@gafter @jaredpar @AlekseyTs @VladimirReshetnikov @VSadov @pharring

### CODE REVIEW

File: Src/Compilers/CSharp/csc2/Csc2.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.CodeAnalysis.BuildTasks;
```
Comment (pharring) [inline Src/Compilers/CSharp/csc2/Csc2.cs]: Please add the OSS header to all files.

File: Src/Compilers/Core/MSBuildTask/BuildClient.cs [hunk 2]
```diff
@@ -48,6 +48,81 @@ private static string GetExpectedServerExeDir()
...
+            }
+
+            var responseTask = TryRunServerCompilation(
+                language,
+                Environment.CurrentDirectory,
+                args,
+                default(CancellationToken),
+                libEnvVariable: Environment.GetEnvironmentVariable("LIB"));
+
+            responseTask.Wait();
```
Comment (gafter) [inline Src/Compilers/Core/MSBuildTask/BuildClient.cs]: Can this throw an AggregateException?

File: Src/Compilers/Core/MSBuildTask/BuildClient.cs [hunk 3]
```diff
@@ -48,6 +48,81 @@ private static string GetExpectedServerExeDir()
...
+
+            int exitCode;
+            var response = responseTask.Result;
```
Comment (gafter) [inline Src/Compilers/Core/MSBuildTask/BuildClient.cs:73]: Can this throw an `AggregateException`?

File: src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs [hunk 4]
```diff
@@ -317,6 +317,45 @@ internal static string RemoveTrailingSpacesAndDots(string path)
...
+        /// Returns an error message if any of the client arguments are invalid
+        /// and null otherwise.
+        /// </summary>
+        internal static string CheckClientArgsForErrors(IEnumerable<string> args)
+        {
+            foreach (var arg in args)
+            {
+                const string keepAlive = "/keepalive";
+                var prefixLength = keepAlive.Length;
+                if (arg.StartsWith(keepAlive))
```
Comment (gafter) [inline src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs]: Is this intentionally case-sensitive?

**Comment by pharring** [comment]
LGTM

File: Src/Compilers/Core/MSBuildTask/BuildClient.cs (hunk 3 above)
Comment (agocke) [inline Src/Compilers/Core/MSBuildTask/BuildClient.cs:73]: It's probably possible. Do you think it's worth trying to put in an exception filter fail fast somewhere here? I don't see a great place to put a `try` block (maybe in TryRunServerCompilation?) -- do you have any suggestions?

**Pull request closed without merging**